.github/workflows/kubeflow-release.yaml

```diff
@@ -0,0 +1,140 @@
+  wait-images-to-build-on-quay:
+    needs: sync-main-to-release-branch
+    runs-on: ubuntu-latest
+    outputs:
+      images_ready: ${{ steps.check-images.outputs.images_ready }}
+    steps:
+      - name: Poll for images availability
+        id: check-images
+        run: |
+          # Install required tools
+          sudo apt-get update
+          sudo apt-get install -y skopeo jq curl
+
+          # Get the latest Git hash from the target branch
+          PAYLOAD=$(curl --silent -H 'Accept: application/vnd.github.v4.raw' https://api.github.com/repos/$REPO_OWNER/$REPO_NAME/commits?sha=$BRANCH_NAME&per_page=1)
+          GIT_HASH=$(echo "$PAYLOAD" | jq -r '.[0].sha' | cut -c 1-7)
+          echo "GIT_HASH=$GIT_HASH"
+
+          # Images to check
+          IMAGES=(
+            "quay.io/opendatahub/kubeflow-notebook-controller:1.9-${GIT_HASH}"
+            "quay.io/opendatahub/odh-notebook-controller:1.9-${GIT_HASH}"
+          )
+          # Poll for image readiness total timeout=15m
+          MAX_ATTEMPTS=10
+          SLEEP_DURATION=90
+          for image in "${IMAGES[@]}"; do
+            for (( i=1; i<=MAX_ATTEMPTS; i++ )); do
+              echo "Checking availability of $image (Attempt $i/$MAX_ATTEMPTS)..."
+              if skopeo inspect docker://$image &>/dev/null; then
+                echo "$image is available!"
+                break
+              fi
+              if [[ $i -eq $MAX_ATTEMPTS ]]; then
+                echo "Timed out waiting for $image to become available."
+                exit 1
+              fi
+              sleep $SLEEP_DURATION
+            done
+          done
+          echo "images_ready=true" >> $GITHUB_ENV
+          echo "images_ready=true" >> $GITHUB_OUTPUT
+
+      - name: Images are ready
+        if: ${{ env.images_ready == 'true' }}
+        run: echo "All images are ready. Proceeding to the next step."
+
+  # 3. Once Images are availble then updates the notebook controllers’ image tags
+  update-release-images:
+    needs: wait-images-to-build-on-quay
+    if: ${{ needs.wait-images-to-build-on-quay.outputs.images_ready == 'true' }}
+    uses: opendatahub-io/kubeflow/.github/workflows/notebook-controller-images-updater.yaml@main
+    with:
+      branch-name: "v1.9-branch"
+      organization: "opendatahub-io"
+      generate-pr: "true"
+
+  # 4. Check PR merged status
+  check-pr-merged:
+    needs: update-release-images
+    runs-on: ubuntu-latest
+    outputs:
+      pr_merged: ${{ steps.check.outputs.pr_merged }}
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Check if the PR is merged
+        id: check
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # PR to look for
+          PR_TITLE="[GHA-${{ github.run_id }}]"
+          # Fetch matching PRs
+          gh pr list --repo atheo89/kubeflow --state all --search "$PR_TITLE" --json number,title > pr_list.json
```

Comment:
We can also use the `--jq` option on the `gh` CLI call to avoid needing to dump this to file.  And while presently we specify in `--json` to return the `title` - I don't see us actually using that anywhere..

Reply:
Updated  json file removed